index.d.ts

```diff
@@ -0,0 +1,39 @@
+declare module 'indexifier' {
+    export interface Node {
+        name: string;
+        children: Node[];
+    }
+  
+    export interface Printer {
+        print(node: Node): string;
+        printNode(node: Node): string;
+    }
+  
+    export type IndexifierOpts = {
+        fileTypes?: string[];
+        isHtml?: boolean;
+        linkFolders?: boolean;
+        include?: RegExp;
+        exclude?: RegExp;
+        emptyDirectories?: boolean;
+        maxDepth?: number;
+        printer?: Printer;
+    };
+  
+    declare function indexifier(dir: string, opts?: IndexifierOpts): string;
+  
+    declare class AbstractPrinter {
+        print(node: Node): string;
+        abstract printNode(node: Node): string;
+    }
+  
+    interface Indexifier {
```

Comment:
```suggestion
    interface Indexifier extends typeof indexifier {
```
maybe?

Reply:
I am not actually sure - isn't that what usually the namespaces are used for? To collapse the two definitions into one? If you have this running in a typescript project with the current types that is good enough for now I think?